### PR TITLE
dev-util/pycharm-professional: add 2024.2.3 | dev-util/pycharm-community: add 2024.2.3

### PR DESCRIPTION
**Pull request by negril**
Note that both ebuilds are identical now. So they are easy to maintain by the same maintainer.

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

File: dev-util/pycharm-community/pycharm-community-2024.2.3.ebuild
Comment (juippis) [inline dev-util/pycharm-community/pycharm-community-2024.2.3.ebuild:13]: Can we sort these (I can do it too while merging though)

File: dev-util/pycharm-community/pycharm-community-2024.2.3.ebuild
Comment (juippis) [inline dev-util/pycharm-community/pycharm-community-2024.2.3.ebuild:63]: None of the licenses imply mirror restriction. Where does this come from?

File: dev-util/pycharm-community/pycharm-community-2024.2.3.ebuild
Comment (juippis) [inline dev-util/pycharm-community/pycharm-community-2024.2.3.ebuild:78]: Can we get rid of the `-f`s to see if there's dead weight being carried?

File: dev-util/pycharm-community/pycharm-community-2024.2.3.ebuild
Comment (juippis) [inline dev-util/pycharm-community/pycharm-community-2024.2.3.ebuild:65]: `${P}` includes a version too, but these are installed to `/opt/${PN}` resultin in hundred lines of `* QA Notice: Files built without respecting CFLAGS/LDFLAGS have been detected` etc.

**Review by juippis — COMMENTED** [review]
Now this is a bit annoying one:
```
 * QA Notice: Unresolved soname dependencies:
 * 
 *      /opt/pycharm-professional/lib/async-profiler/aarch64/libasyncProfiler.so: ld-linux-aarch64.so.1 libc.so.6 libdl.so.2 libgcc_s.so.1 libm.so.6 libpthread.so.0 librt.so.1 libstdc++.so.6
 *      /opt/pycharm-professional/plugins/python-ce/helpers/pydev/pydevd_attach_to_process/attach_linux_aarch64.so: ld-linux-aarch64.so.1 libc.so.6 libgcc_s.so.1 libstdc++.so.6
 *      /opt/pycharm-professional/plugins/remote-dev-server/selfcontained/bin/Xvfb: libaudit.so.1
```
The missing libaudit looks legit. For the others, maybe these aarch64-related files should be deleted on non-arm64 arches?

Also when it comes to deps, there's a bit cleaning to do, but that's not too important to me:
```
 * app-arch/brotli          < 
 * app-arch/zstd            < 
 * app-crypt/p11-kit        < 
 * dev-libs/fribidi         < 
 *                          > dev-libs/expat
 * dev-libs/glib            | dev-libs/glib:2
 * dev-libs/json-c          < 
 * dev-libs/libbsd          < 
 * dev-libs/libdbusmenu     < 
 *                          > dev-libs/nspr
 * media-gfx/graphite2      < 
 * media-libs/fontconfig    < 
 * media-libs/harfbuzz      < 
 * media-libs/libglvnd      < 
 * media-libs/libjpeg-turbo < 
 * media-libs/libpng        < 
 * net-libs/gnutls          < 
 * sys-libs/libcap          < 
 *                          > x11-libs/cairo
 *                          > x11-libs/libXfixes
```

**Comment by juippis** [comment]
Comments apply pretty much to both versions- comunity and professional.

**Comment by negril** [comment]
Yeah I just made the `-community` version work for `-professional`. How do you want to handle this, given that grozin as the maintainer for `-community` only posts to the gentoo-dev ml?

**Comment by juippis** [comment]
I'll merge both if you fix them, let's not throw these away. But I'll merge professional-only too if you don't care about the other. Up to you how much you want to spend time on these :)

File: dev-util/pycharm-professional/metadata.xml
Comment (juippis) [inline dev-util/pycharm-professional/metadata.xml:7]: Is there more to this, I'm looking for the why anyone would ever choose this option?

File: dev-util/pycharm-professional/pycharm-professional-2024.3.ebuild
Comment (juippis) [inline dev-util/pycharm-professional/pycharm-professional-2024.3.ebuild:145]: So this will conflict with xkeyboard-config package that gets pulled by xorg-server and can't be avoided. Therefore `+bundled-xvfb` will always fail to install due to file collisions. 

I imagine *Xvfb* is avoidable, but the ebuild should then have a blocker against `xorg-server[xvfb]`. I'd personally just depend on that instead and forget trying to bundle X with the package. This just... feels weird and unnecessarily complex?

**Review by juippis — COMMENTED** [review]
This thing o.o

File: dev-util/pycharm-professional/pycharm-professional-2024.3.ebuild
Comment (negril) [reply dev-util/pycharm-professional/pycharm-professional-2024.3.ebuild]: The first name is the target. So it would've linked the installed Xvfb into the location of the bundled Xvfb.

**Comment by negril** [comment]
I forgot to push the updated version after removing the draft status. Sorry.

**Review by juippis — APPROVED** [review]
I build-tested this PR but forgot to test running the software. Let's not wait for tomorrow anymore. Thanks!

**Pull request closed without merging**
